internal/db/dqlite.go

```diff
@@ -425,6 +425,28 @@ func dqliteNetworkDial(ctx context.Context, addr string, db *DqliteDB) (net.Conn
 	}
 
 	defer response.Body.Close()
+	if response.StatusCode == http.StatusNotFound {
```

Comment:
The problem is we are reaching out to other systems. Adding an additional request per dqlite connection to grab the target system's API extensions would be quite expensive.

Reply:
Yes sure but isn't this only once during startup?

Reply:
Well, it's continuously run as dqlite rebalances roles until all nodes are updated. 

Additionally, since v1 doesn't support API extensions (or the `/core`) path, it would be equivalent to just sending out a request to `/core/1.0` and checking for a `404` on that instead.

Reply:
Good point with the v1, then pls forget about this discussion.